static/src/javascripts/projects/commercial/modules/messenger/background.js

```diff
@@ -0,0 +1,100 @@
+// @flow
+import { addEventListener } from 'lib/events';
+import fastdom from 'lib/fastdom-promise';
+
+import { register } from 'commercial/modules/messenger';
+
+type AdSpec = {
+    scrollType: string,
+    backgroundColor: string,
+    backgroundImage: string,
+    backgroundRepeat: string,
+    backgroundPosition: string,
+};
+
+const setBackground = (specs: AdSpec, adSlot: Node): Promise<any> | null => {
+    if (
+        !specs ||
+        !('backgroundImage' in specs) ||
+        !('backgroundRepeat' in specs) ||
+        !('backgroundPosition' in specs) ||
+        !('scrollType' in specs)
+    ) {
+        return null;
+    }
+
+    const specsStyles = Object.keys(specs).reduce((result, key) => {
+        if (key !== 'scrollType') {
+            result[key] = specs[key];
+        }
+        // Flow is love, Flow is Life! DFP has been passing us `backgroundColour`
+        // all along, and the setting of this css prop has been silently failing
+        if (key === 'backgroundColour') {
+            result.backgroundColor = specs[key];
+        }
+        return result;
+    }, {});
+
+    // Create an element to hold the background image
+    const background = document.createElement('div');
+    background.className = `creative__background creative__background--${specs.scrollType}`;
+    background.style = Object.assign(background.style, specsStyles);
```

Comment:
Ah, I see, it's because `backgroundColour` is not specified in the `AdSpec` type definition. Should we update that to expect `backgroundColour` instead of `backgroundColor`?